test/test-mc-text-search-str-encode.c

```diff
@@ -1198,10 +1199,129 @@ static void _test_text_search_str_encode_empty_string(_mongocrypt_tester_t *test
     }
 }
 
+// Tests mc_text_search_str_query() fails on invalid utf-8.
+static void _test_text_search_str_query_bad_string(_mongocrypt_tester_t *tester) {
+    mongocrypt_status_t *status = mongocrypt_status_new();
+    _mongocrypt_buffer_t out;
+    mc_FLE2TextSearchInsertSpec_t spec = {.v = "\xff\xff\xff\xff\xff\xff\xff\xff\xff", .len = 9};
+
+    bool res = mc_text_search_str_query(&spec, &out, status);
+    ASSERT_FAILS_STATUS(res, status, "not valid UTF-8");
+    mongocrypt_status_destroy(status);
+}
+
+// Tests mc_text_search_str_query() checks the input string codepoint length against lb and ub
+// for substring, suffix, and prefix queries, and not for exact queres.
+static void _test_text_search_str_query_bounds_checking(_mongocrypt_tester_t *tester) {
+    mc_FLE2TextSearchInsertSpec_t substrSpec = {.substr = {{20, 4, 7}, true}};
+    mc_FLE2TextSearchInsertSpec_t suffixSpec = {.suffix = {{4, 7}, true}};
+    mc_FLE2TextSearchInsertSpec_t prefixSpec = {.prefix = {{4, 7}, true}};
+    mc_FLE2TextSearchInsertSpec_t *specs[3] = {&substrSpec, &suffixSpec, &prefixSpec};
+
+    char *short_str = build_random_string_to_fold(3, 5);
+    char *long_str = build_random_string_to_fold(8, 10);
+
+    mongocrypt_status_t *status = mongocrypt_status_new();
+    _mongocrypt_buffer_t out;
+
+    for (int i = 0; i < 3; i++) {
+        // long_str always fails regardless of folding
+        for (int d = 0; d < 2; d++) {
+            specs[i]->v = long_str;
+            specs[i]->len = strlen(specs[i]->v);
```

Comment:
```suggestion
            specs[i]->len = (uint32_t) strlen(specs[i]->v);
```

To address task failures on Windows ([example](https://spruce.mongodb.com/task/libmongocrypt_windows_test_build_and_test_and_upload_patch_c15c6f06cb5e3ee12146992e96619e6e0f63c787_67eb0089955418000742c65e_25_03_31_20_52_25/logs?execution=0)). Similar suggestion applies to other cases.